.github/workflows/release.yml

```
@@ -7,12 +7,89 @@ on:
       version:
         description: "Specify the version"
         required: true
-        default: "v1.0.0"
+        default: "v0.13.1"
   pull_request:
     branches:
       - "master"
 
+permissions:
+  contents: "write"
+
+env:
+  TMKMS_VERSION: "${{ inputs.version || 'v0.13.1' }}"
+
 jobs:
   build:
     name: "Build"
-    runs-on: "ubuntu-latest-l"
\ No newline at end of file
+    runs-on: "ubuntu-latest-l"
+    steps:
+      - id: "setup-rust"
+        name: "Set up Rust toolchain"
+        uses: "dtolnay/rust-toolchain@v1"
+        with:
+          toolchain: "stable"
+
+      - id: "setup-dependencies"
+        name: "Install required system packages"
+        run: |
+          sudo apt-get update
+          sudo apt-get install -y --no-install-recommends build-essential curl git musl-dev
+
+      - id: "clone"
+        name: "Clone tmkms source"
+        uses: "actions/checkout@v4"
+        with:
+          repository: "iqlusioninc/tmkms"
+          fetch-tags: true
+          path: "tmkms"
+          ref: "refs/tags/${{ env.TMKMS_VERSION }}"
+          submodules: true
+
+      - id: "cache"
+        name: "Cache Cargo dependencies"
+        uses: "actions/cache@v3"
+        with:
+          path: "~/.cargo"
+          key: "cargo-${{ hashFiles('tmkms/Cargo.lock') }}"
```

```suggestion
          key: "cargo-${{ hashFiles('tmkms/Cargo.lock') }}"
          restore-keys: |
            cargo-${{ hashFiles('tmkms/Cargo.lock') }}
            cargo-
```

See https://docs.github.com/en/actions/using-workflows/caching-dependencies-to-speed-up-workflows#example-using-the-cache-action